src/odemis/acq/stream/_sync.py

```diff
@@ -1303,19 +1307,23 @@ def _runAcquisitionEbeam(self, future):
                             self._sccd._onNewData(self._ccd_df, self._acq_data[self._ccd_idx][-1])
                         except Exception:
                             logging.exception("Failed to update CCD live view")
+                        # check if we need to integrate images (integration count > 1)
+                        if integration_count > 1:
+                            for stream_idx, das in enumerate(self._acq_data):
+                                if self._img_intor[stream_idx] is None:
+                                    self._img_intor[stream_idx] = img.ImageIntegrator(integration_count)
+                                self._acq_data[stream_idx] = self._img_intor[stream_idx].append(das)
```

Comment:
Just pass the "last" and only DataArray in das => `self._acq_data[stream_idx][-1] = ... .append(das[-1])`.